backend/tests/models/pipeline/BuildInfo.test.ts

```diff
@@ -0,0 +1,131 @@
+  const pipielineStep = "Deploy Integration App";
+  const bkJobInfo: BKJobInfo = {
+    name: "Deploy Integration App",
+    state: "passed",
+    startedAt: "2021-12-16T22:10:29.122Z",
+    finishedAt: "2021-12-16T22:10:58.849Z",
+  };
+  const bkBuildInfo: BKBuildInfo = {
+    jobs: [bkJobInfo],
+    commit: "12",
+    pipelineCreateTime: "2021-12-17T02:11:55.965Z",
+    number: 9400,
+  };
+  const buildInfo = new BuildInfo(bkBuildInfo);
+  describe("mapToDeployInfo", () => {
+    it("should return a corresponding deployinfo when having a job matching step and states", () => {
+      const result = buildInfo.mapToDeployInfo(
+        pipielineStep,
+        "passed",
+        "failed"
+      );
+      const expected = new DeployInfo(
+        "2021-12-17T02:11:55.965Z",
+        "2021-12-16T22:10:29.122Z",
+        "2021-12-16T22:10:58.849Z",
+        "12",
+        "passed"
+      );
+      expect(result).deep.equal(expected);
+    });
+    it("should return a blank deployinfo when not having pipeline create time", () => {
+      const bkBuildInfoWithoutPipelineCreateTime: BKBuildInfo = {
+        ...bkBuildInfo,
+        pipelineCreateTime: undefined,
+      };
+      const buildInfo = new BuildInfo(bkBuildInfoWithoutPipelineCreateTime);
+      const result = buildInfo.mapToDeployInfo(
+        pipielineStep,
+        "passed",
+        "failed"
+      );
+      const expected = new DeployInfo("", "", "", "", "");
+      expect(result).deep.equal(expected);
+    });
+    it("should return a blank deployinfo when having no job matching step", () => {
+      const pipielineStepNotMatched = "deploy production App";
+      const result = buildInfo.mapToDeployInfo(
+        pipielineStepNotMatched,
+        "passed",
+        "failed"
+      );
+      const expected = new DeployInfo("", "", "", "", "");
+      expect(result).deep.equal(expected);
+    });
+    it("should return a blank deployinfo when having job matching step but not states", () => {
+      const pipielineStepNotMatched = "deploy production App";
+      const result = buildInfo.mapToDeployInfo(
+        pipielineStepNotMatched,
+        "running",
+        "failed"
+      );
+      const expected = new DeployInfo("", "", "", "", "");
+      expect(result).deep.equal(expected);
+    });
+    it("should return a blank deployinfo when matched job without start time", () => {
+      const bkJobInfoWithoutStartTime: BKJobInfo = {
+        name: "Deploy Integration App",
+        state: "passed",
+        finishedAt: "2021-12-16T22:10:58.849Z",
+      };
+      const bkBuildInfo: BKBuildInfo = {
+        jobs: [bkJobInfoWithoutStartTime],
+        commit: "12",
+        pipelineCreateTime: "2021-12-17T02:11:55.965Z",
+        number: 9400,
+      };
+      const buildInfo = new BuildInfo(bkBuildInfo);
+
+      const result = buildInfo.mapToDeployInfo(
+        pipielineStep,
+        "passed",
+        "failed"
+      );
+      const expected = new DeployInfo("", "", "", "", "");
```

Comment:
It looks like the object `expected = new DeployInfo("", "", "", "", "");` is used in many places, so maybe it could be defined at the beginning and used directly elsewhere

Reply:
it has been changed now